pkg/syncer/pv_to_vvol_mapping.go

```diff
@@ -0,0 +1,175 @@
+		updated, err := updatePVtoBackingDiskObjectIdMappingStatus(ctx, k8sclient, pvc, pvToBackingDiskObjectIdPair)
+		if err != nil {
+			log.Error("csiGetPVtoBackingDiskObjectIdMapping: Failed to update pv to backingDiskObjectId mapping")
+			return
+		}
+		if updated {
+			log.Debug("csiGetPVtoBackingDiskObjectIdMapping: pv to backingDiskObjectId mapping updated")
+		}
+	}
+
+	log.Debug("csiGetPVtoBackingDiskObjectIdMapping: end")
+}
+
+func parseMappingString(pvTovvolPair string, ctx context.Context) (string, string, error) {
+	log := logger.GetLogger(ctx)
+	split := strings.Split(pvTovvolPair, ":")
+	if len(split) != 2 {
+		log.Error("Failed to parse mapping string")
+		return "", "", errors.New("failed to parse mapping string")
+	}
+	return split[0], split[1], nil
+}
+
+func updatePVtoBackingDiskObjectIdMappingStatus(ctx context.Context, k8sclient clientset.Interface,
+	pvc *v1.PersistentVolumeClaim, pvToBackingDiskObjectIdPair string) (bool, error) {
+	log := logger.GetLogger(ctx)
+
+	uuid, backingDiskObjectId, err := parseMappingString(pvToBackingDiskObjectIdPair, ctx)
+	if err != nil {
+		log.Errorf("updatePVtoBackingDiskObjectIdMappingStatus: Failed to parse mapping pair for pvc %s/%s with err:%+v",
+			pvc.Namespace, pvc.Name, err)
+		return false, err
+	}
+	val, found := pvc.Annotations[annPVtoBackingDiskObjectId+"/"+uuid]
+
+	if !found || val != backingDiskObjectId {
+		// PVToBackingDiskObjectId annotation on pvc is changed, set it to new value.
+		metav1.SetMetaDataAnnotation(&pvc.ObjectMeta, annPVtoBackingDiskObjectId+"/"+uuid, backingDiskObjectId)
+
+		log.Infof("updatePVtoBackingDiskObjectIdMappingStatus: set pv to vvol annotation for pvc %s/%s from old "+
+			"value %s to new value %s",
+			pvc.Namespace, pvc.Name, val, backingDiskObjectId)
+		_, err := k8sclient.CoreV1().PersistentVolumeClaims(pvc.Namespace).Update(ctx, pvc, metav1.UpdateOptions{})
+		if err != nil {
+			if apierrors.IsConflict(err) {
+				log.Debugf("updatePVtoBackingDiskObjectIdMappingStatus: Failed to update pvc %s/%s with err:%+v, will retry the update",
+					pvc.Namespace, pvc.Name, err)
+				// pvc get from pvcLister may be stale, try to get updated pvc which
+				// bound to pv from API server.
+				newPvc, err := k8sclient.CoreV1().PersistentVolumeClaims(pvc.Namespace).Get(
+					ctx, pvc.Name, metav1.GetOptions{})
+				if err != nil {
+					log.Errorf("updatePVtoBackingDiskObjectIdMappingStatus: pv to vvol annotation for pvc %s/%s is not updated because "+
+						"failed to get pvc from API server. err=%+v",
+						pvc.Namespace, pvc.Name, err)
+					return false, err
+				}
+
+				log.Infof("updatePVtoBackingDiskObjectIdMappingStatus: updating pv to vvol annotation for pvc %s/%s which "+
+					"get from API server from old value %s to new value %s",
+					newPvc.Namespace, newPvc.Name, val, backingDiskObjectId)
+				metav1.SetMetaDataAnnotation(&newPvc.ObjectMeta, annPVtoBackingDiskObjectId+"/"+uuid, backingDiskObjectId)
```

Comment:
See this comment:
https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/1516/files#r799883807